docs/CharmsInDetail.md

```diff
@@ -0,0 +1,292 @@
+# Charms in detail
+## Table of contents
+
+[Best practices](##Best%20practice)
+
+[Charm writing in detail](##Charm%20Writing%20in%20detail)
+
+[New to charms?](##New%20to%20charms?)
+
+[Charm Writing](#Charm%20writing)
+
+[Example Charms](##Example%20Charms)
+
+## Best practices
+
+### Repoository Naming
+
+Naming follows the convention of charm-<charm_name> for example `charm-percona-cluster`.
+
+The rationale behind this is quick and easy searching of the 1000s of github projects along with other reasons.
+
+
+## Charm Writing in detail
+
+## New to charms?
+
+> Here are some completely reasonable questions to be asking right now
+
+### Why charms?
+
+Charms have a noble goal: take the complication of managing 100s potentially 1000s of configuration files and abstract them as python modules. Thus by abstraction creating a true insfrastructure as code.
```

Comment:
"noble" goal. :-) Do we really have to qualify "goal" with that?

Reply:
probably not, i'll remove